main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"code-review-bot/internal"
+	"code-review-bot/internal/web"
+	"code-review-bot/internal/web/routes"
+	"context"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v68/github"
+	"github.com/gregjones/httpcache"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+	"go.uber.org/zap"
+)
+
+func initializeGithubClient(ctx context.Context, githubToken string, logger *zap.Logger) *github.Client {
+	client := github.NewClient(httpcache.NewMemoryCacheTransport().Client()).WithAuthToken(githubToken)
+
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		logger.Fatal("Failed to authenticate with GitHub API", zap.Error(err))
+	}
+	logger.Info("Authenticated with GitHub API", zap.String("username", user.GetLogin()))
+
+	return client
+}
+
+func initializeOpenAIClient(openAIToken string, logger *zap.Logger) *openai.Client {
+	client := openai.NewClient(option.WithAPIKey(openAIToken))
+
+	logger.Info("OpenAI client initialized")
+
+	return client
+}
+
+func main() {
+	logger, _ := zap.NewDevelopment()
+	defer logger.Sync()
+	logger.Info("starting server")
+
+	config, err := internal.LoadConfiguration()
+	if err != nil {
+		logger.Fatal("failed to load configuration", zap.Error(err))
+	}
+
+	e := echo.New()
+	ctx := context.Background()
+	//e.Use(middleware.Recover())
```

Comment:
The GitHub client is initialized without a check for a valid GitHub token. This can lead to potential logical errors if the token is invalid or missing. Consider adding validation for the GitHub token before initializing the client.